backend/logic/payments/stripe.js

```diff
@@ -1,13 +1,142 @@
+const get = require('lodash/get')
 const Stripe = require('stripe')
+
 const { getLogger } = require('../../utils/logger')
-const { ExternalPayment } = require('../../models')
+const { ExternalPayment, Order } = require('../../models')
 const { Sentry } = require('../../sentry')
+const { OrderPaymentTypes } = require('../../enums')
 
 const log = getLogger('utils.stripe')
 
 const { IS_TEST } = require('../../utils/const')
 const { getConfig } = require('../../utils/encryptedConfig')
 
+/**
+ * Check whether a shop is properly configured for Stripe or not.
+ *
+ * @param {models.Network} network
+ * @param {models.Shop} shop
+ * @returns {Promise<{success: false, error: string}|{success: true}>}
+ */
+async function checkStripeConfig(network, shop) {
+  const networkConfig = getConfig(network.config)
+  const validWebhhokUrls = [
+    `${networkConfig.backendUrl}/webhook`,
+    `https://dshopapi.ogn.app/webhook` // Legacy URL. Still supported.
+  ]
+
+  log.info(`Shop {shop.id} - Checking Stripe config`)
+  let success = false
+  try {
+    const shopConfig = getConfig(shop.config)
+    if (!shopConfig.stripeBackend) {
+      log.info(`Shop {shop.id} - Stripe not configured`)
+      return { success: true }
+    }
+    const stripe = Stripe(shopConfig.stripeBackend)
+    const response = await stripe.webhookEndpoints.list()
+    const webhooks = response.data
+
+    for (const webhook of webhooks) {
+      if (validWebhhokUrls.includes(webhook.url)) {
+        log.info(
+          'Shop {shop.id} - Webhook properly configured. Pointing to',
+          webhook.url
+        )
+        success = true
+        break
+      } else {
+        log.warn(
+          `Shop {shop.id} - Webhook id ${webhook.id} points to non-Dshop or invalid URL ${webhook.url}`
+        )
+      }
+    }
+  } catch (e) {
+    log.error(`Shop {shop.id} - Failed checking webhook config: ${e}`)
+  }
+
+  if (!success) {
+    return { success: false, error: 'Stripe Webhook not properly configured' }
+  }
+  return { success: true }
+}
+
+/**
+ * Loads the Stripe payment in the last 30 days for a shop and
+ * checks there is an associated order for each of them in the DB.
+ *
+ * @param {models.Shop} shop
+ * @returns {Promise<{success: false, errors: array<string>}|{success: true}>}
+ */
+async function checkStripePayments(shopId, shopConfig) {
+  const stripeKey = shopConfig.stripeBackend
+  if (!stripeKey) {
+    log.info('No stripe key configured')
+    return
+  }
+  const stripe = Stripe(stripeKey)
+
+  // Load the last 100 shop's orders paid with Stripe and get their encrypted IPFS hashes.
+  const orders = await Order.findAll({
+    where: { shopId, paymentType: OrderPaymentTypes.Stripe },
+    limit: 100,
```

Comment:
The doc above says every order in the last 30 days... should we query based on createdAt and remove the limit? Also perhaps limit attributes to `encryptedIpfsHash` since that's the only field we need? `attributes: ['encryptedIpfsHash']`

Reply:
Done.